### PR TITLE
Sort facilities alphabetically by name

### PR DESCRIPTION
**Pull request by Mitch-A6**
## Summary

- 10-10 EZ form: sort the Facilities list by name in alphabetical order instead of by Facility ID
- 10-10 Health Apps - EZ/CG team

## Related issue(s)

- https://app.zenhub.com/workspaces/10-10-health-apps-5fff0cfd1462b6000e320fc7/issues/gh/department-of-veterans-affairs/va.gov-team/85772

## Testing done

- Facilities were ordered by their (hidden) ID value, (e.g., `520GA`)
- We sort the Facilities list by the Name attribute of each entry prior to display (e.g., `Mobile VA Clinic`)
- Tested locally and in unit tests to confirm that the order is now alphabetical

## Screenshots
before: 
Screenshot 2024-06-12 at 3 56 19 PM

after:
Screenshot 2024-06-12 at 4 01 07 PM

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

**Comment by Mitch-A6** [comment]
As mentioned previously, the warning about missing a dependency on `localData` is incorrect - if we did that, it would unnecessarily reload every time we switched Facilites, too.

File: src/applications/hca/tests/unit/components/FormFields/VaMedicalCenter.unit.spec.js [hunk 1]
```diff
@@ -182,7 +184,8 @@ describe('hca <VaMedicalCenter>', () => {
         const alert = container.querySelector('va-alert');
 
         expect(options).to.have.lengthOf(mockData.length);
-        expect(options[0]).to.have.attr('value', mockData[0].uniqueId);
+        expect(options[0]).to.have.attr('value', mockData[1].uniqueId);
```
Comment (coope93) [inline src/applications/hca/tests/unit/components/FormFields/VaMedicalCenter.unit.spec.js:187]: I'd probably like a comment or update the `it` text to show that we expect these to be alphabetized?

**Review by coope93 — DISMISSED** [review]

File: src/applications/hca/tests/unit/components/FormFields/VaMedicalCenter.unit.spec.js (hunk 1 above)
Comment (Mitch-A6) [reply src/applications/hca/tests/unit/components/FormFields/VaMedicalCenter.unit.spec.js:187]: How's that rename for the test look?

**Review by mkerns1 — APPROVED** [review]
LGTM!

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Review by coope93 — APPROVED** [review]

**Pull request merged**